### Importing Packages

In [162]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.metrics.pairwise import cosine_similarity

### Loading the dataset

- Importing Dataset named as: netflix.csv 

In [163]:
df = pd.read_csv('netflix.csv')

- Getting the information columns that we have and checking their data-types and number instances on which columns

In [164]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


- printing First 5 instances for the reference of dataset

In [165]:
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

- We are checking for the shape which has 8807 instances per column and 12 features 

In [112]:
df.shape

(8807, 12)

- Checking for the dtypes of columns

In [113]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

- Checking for the null values for each columns

In [114]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

- Here we are printing rating column's unique value which would be our movie or show ratings.

In [115]:
print('Types of ratings:',df['rating'].unique())

Types of ratings: ['PG-13' 'TV-MA' 'PG' 'TV-14' 'TV-PG' 'TV-Y' 'TV-Y7' 'R' 'TV-G' 'G'
 'NC-17' '74 min' '84 min' '66 min' 'NR' nan 'TV-Y7-FV' 'UR']


- In this snnipet here we are searching for the total number of movies and their unique durations. 

In [116]:
print('Number of Movies:',
      df[df['type'] == 'Movie']['duration'].count(),
      '\nShow Durations:',
      df[df['type'] == 'Movie']['duration'].unique())

Number of Movies: 6128 
Show Durations: ['90 min' '91 min' '125 min' '104 min' '127 min' '67 min' '94 min'
 '161 min' '61 min' '166 min' '147 min' '103 min' '97 min' '106 min'
 '111 min' '110 min' '105 min' '96 min' '124 min' '116 min' '98 min'
 '23 min' '115 min' '122 min' '99 min' '88 min' '100 min' '102 min'
 '93 min' '95 min' '85 min' '83 min' '113 min' '13 min' '182 min' '48 min'
 '145 min' '87 min' '92 min' '80 min' '117 min' '128 min' '119 min'
 '143 min' '114 min' '118 min' '108 min' '63 min' '121 min' '142 min'
 '154 min' '120 min' '82 min' '109 min' '101 min' '86 min' '229 min'
 '76 min' '89 min' '156 min' '112 min' '107 min' '129 min' '135 min'
 '136 min' '165 min' '150 min' '133 min' '70 min' '84 min' '140 min'
 '78 min' '64 min' '59 min' '139 min' '69 min' '148 min' '189 min'
 '141 min' '130 min' '138 min' '81 min' '132 min' '123 min' '65 min'
 '68 min' '66 min' '62 min' '74 min' '131 min' '39 min' '46 min' '38 min'
 '126 min' '155 min' '159 min' '137 min' '12 min' '273 mi

- Same we are doing the shows that are presented inside our dataset

In [117]:
print('Number of Shows:',
      df[df['type'] == 'TV Show']['duration'].count(),
      '\nShow Durations:',
      df[df['type'] == 'TV Show']['duration'].unique())

Number of Shows: 2676 
Show Durations: ['2 Seasons' '1 Season' '9 Seasons' '4 Seasons' '5 Seasons' '3 Seasons'
 '6 Seasons' '7 Seasons' '10 Seasons' '8 Seasons' '17 Seasons'
 '13 Seasons' '15 Seasons' '12 Seasons' '11 Seasons']


<h4>Observations</h4>

<ul>
    <li>A lot of null values, especially for director, cast and country</li>
    <li>Rating isn't the popularity, but the suggested audience of a movie/film</li>
    <li>Listed In is essntially the genre list</li>
    <li>Duration is kept in two different metrics, time for movies and seasons for shows</li>
</ul>

# Preprocessing & Visualization

<h3>General</h3>

- Dropping Director, Cast, and Country, due to the amoung of null values and how unlikely it is that they will have a significant effect on future models

In [118]:
df = df.drop(['director', 'cast', 'country'], axis=1)

- Dropping duration column, as two different metrics are used depending on whether the item is a show or a movie. Another solution would be to split the dataset into movies and shows, but we want to be able to recommend either.

In [119]:
df = df.drop('duration', axis=1)

- Dropping date added, since it has little relevance

In [120]:
df = df.drop('date_added', axis=1)

In [121]:
df.isnull().sum()

show_id         0
type            0
title           0
release_year    0
rating          4
listed_in       0
description     0
dtype: int64

- Renaming two-word column names for ease

In [122]:
df = df.rename({'release_year':'year'}, axis=1)
df = df.rename({'listed_in':'category'}, axis=1)
df = df.rename({'show_id':'id'}, axis=1)

In [123]:
df[df['rating'].isnull()].head()

id     type                                              title  year  \
5989  s5990    Movie  13TH: A Conversation with Oprah Winfrey & Ava ...  2017   
6827  s6828  TV Show                  Gargantia on the Verdurous Planet  2013   
7312  s7313  TV Show                                       Little Lunch  2015   
7537  s7538    Movie                               My Honor Was Loyalty  2015   

     rating                              category  \
5989    NaN                                Movies   
6827    NaN  Anime Series, International TV Shows   
7312    NaN                 Kids' TV, TV Comedies   
7537    NaN                                Dramas   

                                            description  
5989  Oprah Winfrey sits down with director Ava DuVe...  
6827  After falling through a wormhole, a space-dwel...  
7312  Adopting a child's perspective, this show take...  
7537  Amid the chaos and horror of World War II, a c...

- Looking at the remaining null values, which is the four null ratings, it is low enough that we can drop those in order to simplify the computations being done

In [124]:
df = df.dropna()

In [125]:
df.shape

(8803, 7)

<h3>RS Processing</h3>

- One hot encoding the features, preparing dataframe for recommendation system

- Different from our first iteration, we are going to use rating and type as well

<h4>First, the ratings</h4>

In [126]:
rating_enc = pd.get_dummies(df['rating'], prefix='rating')

In [127]:
rating_enc.columns

Index(['rating_66 min', 'rating_74 min', 'rating_84 min', 'rating_G',
       'rating_NC-17', 'rating_NR', 'rating_PG', 'rating_PG-13', 'rating_R',
       'rating_TV-14', 'rating_TV-G', 'rating_TV-MA', 'rating_TV-PG',
       'rating_TV-Y', 'rating_TV-Y7', 'rating_TV-Y7-FV', 'rating_UR'],
      dtype='object')

In [128]:
rating_enc = rating_enc.drop(['rating_66 min', 'rating_74 min', 'rating_84 min'], axis=1)

In [129]:
rating_enc = rating_enc.drop(['rating_NR','rating_UR'], axis=1)

In [130]:
rating_enc = rating_enc.rename({'rating_PG':'Rated PG',
                                'rating_R':'Rated R',
                                'rating_TV-14':'Rated TV_14',
                                'rating_TV-G':'Rated TV_G',
                                'rating_TV-MA':'Rated TV_MA',
                                'rating_TV-PG':'Rated TV_PG',
                                'rating_TV-Y':'Rated TV_Y',
                                'rating_TV-Y7':'Rated TY_Y7',
                                'rating_TV-Y7-FV':'Rated TV_Y7_FV',
                                'rating_G':'Rated G',
                                'rating_NC-17':'Rated NC-17'}, axis=1)

In [131]:
rating_enc.columns

Index(['Rated G', 'Rated NC-17', 'Rated PG', 'rating_PG-13', 'Rated R',
       'Rated TV_14', 'Rated TV_G', 'Rated TV_MA', 'Rated TV_PG', 'Rated TV_Y',
       'Rated TY_Y7', 'Rated TV_Y7_FV'],
      dtype='object')

<h4>Next, the type</h4>

In [132]:
type_enc = pd.get_dummies(df['type'], prefix='type')

In [133]:
type_enc.columns

Index(['type_Movie', 'type_TV Show'], dtype='object')

In [134]:
type_enc = type_enc.rename({'type_Movie':'Movie',
                            'type_TV Show':'Show'}, axis=1)

<h4>Lastly, the genre of the movie/show</h4>

In [135]:
genre_enc = df.filter('category')
genre_enc['category'] = df['category'].str.replace(' ', '')
genre_enc = genre_enc.join(pd.concat([genre_enc['category'].str.get_dummies(sep=',')])).drop('category', axis=1)

In [136]:
genre_enc.columns

Index(['Action&Adventure', 'AnimeFeatures', 'AnimeSeries', 'BritishTVShows',
       'Children&FamilyMovies', 'Classic&CultTV', 'ClassicMovies', 'Comedies',
       'CrimeTVShows', 'CultMovies', 'Documentaries', 'Docuseries', 'Dramas',
       'Faith&Spirituality', 'HorrorMovies', 'IndependentMovies',
       'InternationalMovies', 'InternationalTVShows', 'Kids'TV',
       'KoreanTVShows', 'LGBTQMovies', 'Movies', 'Music&Musicals', 'RealityTV',
       'RomanticMovies', 'RomanticTVShows', 'Sci-Fi&Fantasy',
       'Science&NatureTV', 'Spanish-LanguageTVShows', 'SportsMovies',
       'Stand-UpComedy', 'Stand-UpComedy&TalkShows', 'TVAction&Adventure',
       'TVComedies', 'TVDramas', 'TVHorror', 'TVMysteries', 'TVSci-Fi&Fantasy',
       'TVShows', 'TVThrillers', 'TeenTVShows', 'Thrillers'],
      dtype='object')

- Noticing a lot of redundancy between categories, especially to split movies and shows. I am going to essentially merge the two

In [137]:
pd.set_option('display.max_columns', None)
genre_enc

Action&Adventure  AnimeFeatures  AnimeSeries  BritishTVShows  \
0                    0              0            0               0   
1                    0              0            0               0   
2                    0              0            0               0   
3                    0              0            0               0   
4                    0              0            0               0   
...                ...            ...          ...             ...   
8802                 0              0            0               0   
8803                 0              0            0               0   
8804                 0              0            0               0   
8805                 0              0            0               0   
8806                 0              0            0               0   

      Children&FamilyMovies  Classic&CultTV  ClassicMovies  Comedies  \
0                         0               0              0         0   
1                         0               0              0         0   
2                         0               0              0         0   
3                         0               0              0         0   
4                         0               0              0         0   
...                     ...             ...            ...       ...   
8802                      0               0              0         0   
8803                      0               0              0         0   
8804                      0               0              0         1   
8805                      1               0              0         1   
8806                      0               0              0         0   

      CrimeTVShows  CultMovies  Documentaries  Docuseries  Dramas  \
0                0           0              1           0       0   
1                0           0              0           0       0   
2                1           0              0           0       0   
3                0           0              0           1       0   
4                0           0              0           0       0   
...            ...         ...            ...         ...     ...   
8802             0           1              0           0       1   
8803             0           0              0           0       0   
8804             0           0              0           0       0   
8805             0           0              0           0       0   
8806             0           0              0           0       1   

      Faith&Spirituality  HorrorMovies  IndependentMovies  \
0                      0             0                  0   
1                      0             0                  0   
2                      0             0                  0   
3                      0             0                  0   
4                      0             0                  0   
...                  ...           ...                ...   
8802                   0             0                  0   
8803                   0             0                  0   
8804                   0             1                  0   
8805                   0             0                  0   
8806                   0             0                  0   

      InternationalMovies  InternationalTVShows  Kids'TV  KoreanTVShows  \
0                       0                     0        0              0   
1                       0                     1        0              0   
2                       0                     1        0              0   
3                       0                     0        0              0   
4                       0                     1        0              0   
...                   ...                   ...      ...            ...   
8802                    0                     0        0              0   
8803                    0                     0        1              1   
8804                    0                     0        0       

- Getting the first row AnimeFeatures and from here I will be assinging the repsective column to 1 for the encoding puprose

- Here we are renaming the AnimeFeatures to Anime and placing it where AnimeSeries to 1. I will be repeating this thing for every categories or movie and shows

In [138]:
genre_enc = genre_enc.rename({'AnimeFeatures':'Anime'}, axis=1)
genre_enc.loc[genre_enc['AnimeSeries'] == 1, 'Anime'] = 1
genre_enc = genre_enc.drop('AnimeSeries', axis=1)

<h5>Classical and Cult</h5>

- Spilling Classic and Cult into two, assigning Classic&Cult to both, then comboning classicTV and classicMovies

In [139]:

genre_enc = genre_enc.rename({'Classic&CultTV':'Classic'}, axis=1)

genre_enc = genre_enc.rename({'CultMovies':'Cult'}, axis=1)
genre_enc.loc[genre_enc['Classic'] == 1, 'Cult'] = 1

genre_enc.loc[genre_enc['ClassicMovies'] == 1, 'Classic'] = 1
genre_enc = genre_enc.drop('ClassicMovies', axis=1)

<h5>Documentaries</h5>

In [140]:
genre_enc.loc[genre_enc['Docuseries'] == 1, 'Documentaries'] = 1
genre_enc = genre_enc.drop('Docuseries', axis=1)

<h5>Family/Kids</h5>

In [141]:
# Essentially, splitting Family and Children/Kids categories and adding Children&Family to both
genre_enc = genre_enc.rename({'Children&FamilyMovies':'Family'}, axis=1)
genre_enc = genre_enc.rename({"Kids'TV":'Kids'}, axis=1)

genre_enc.loc[genre_enc['Family'] == 1, 'Kids'] = 1

<h5>International</h5>

In [142]:
genre_enc = genre_enc.rename({'InternationalMovies':'International'}, axis=1)
genre_enc.loc[genre_enc['InternationalTVShows'] == 1, 'International'] = 1
genre_enc = genre_enc.drop('InternationalTVShows', axis=1)

Puts stand-up comedy into talk shows and comedies before dropping the column

In [143]:

genre_enc = genre_enc.rename({'Comedies':'Comedy'}, axis=1)

genre_enc = genre_enc.rename({'Stand-UpComedy&TalkShows':'TalkShow'}, axis=1)
genre_enc.loc[genre_enc['Stand-UpComedy'] == 1, 'TalkShow'] = 1

genre_enc.loc[genre_enc['Stand-UpComedy'] == 1, 'Comedy'] = 1
genre_enc = genre_enc.drop('Stand-UpComedy', axis=1)

<h5>Misc. Renaming</h5>
- Here I am renaming and assigning 1 to repsective column for the encoding purpose

In [144]:
genre_enc = genre_enc.rename({'BritishTVShows':'British'}, axis=1)
genre_enc = genre_enc.rename({'CrimeTVShows':'Crime'}, axis=1)
genre_enc = genre_enc.rename({'Faith&Spirituality':'Faith'}, axis=1)
genre_enc = genre_enc.rename({'IndependentMovies':'Independant'}, axis=1)
genre_enc = genre_enc.rename({'KoreanTVShows':'Korean'}, axis=1)
genre_enc = genre_enc.rename({'LGBTQMovies':'LGBTQ'}, axis=1)
genre_enc = genre_enc.rename({'Music&Musicals':'Musical'}, axis=1)
genre_enc = genre_enc.rename({'RealityTV':'Reality'}, axis=1)
genre_enc = genre_enc.rename({'Science&NatureTV':'Science'}, axis=1)
genre_enc = genre_enc.rename({'Spanish-LanguageTVShows':'Spanish'}, axis=1)
genre_enc = genre_enc.rename({'SportsMovies':'Sports'}, axis=1)
genre_enc = genre_enc.rename({'TVMysteries':'Mystery'}, axis=1)
genre_enc = genre_enc.rename({'TeenTVShows':'Teen'}, axis=1)

In [145]:
genre_enc = genre_enc.drop('Movies', axis=1)
genre_enc = genre_enc.drop('TVShows', axis=1)

In [146]:
genre_enc.loc[genre_enc['TVDramas'] == 1, 'Dramas'] = 1
genre_enc = genre_enc.drop('TVDramas', axis=1)

In [147]:
genre_enc = genre_enc.rename({'Action&Adventure':'Action'}, axis=1)
genre_enc.loc[genre_enc['TVAction&Adventure'] == 1, 'Action'] = 1
genre_enc = genre_enc.drop('TVAction&Adventure', axis=1)

In [148]:
genre_enc = genre_enc.rename({'RomanticMovies':'Romance'}, axis=1)
genre_enc.loc[genre_enc['RomanticTVShows'] == 1, 'Romance'] = 1
genre_enc = genre_enc.drop('RomanticTVShows', axis=1)

In [149]:
genre_enc = genre_enc.rename({'HorrorMovies':'Horror'}, axis=1)
genre_enc.loc[genre_enc['TVHorror'] == 1, 'Horror'] = 1
genre_enc = genre_enc.drop('TVHorror', axis=1)

In [150]:
genre_enc = genre_enc.rename({'Sci-Fi&Fantasy':'SciFiFan'}, axis=1)
genre_enc.loc[genre_enc['TVSci-Fi&Fantasy'] == 1, 'SciFiFan'] = 1
genre_enc = genre_enc.drop('TVSci-Fi&Fantasy', axis=1)

In [151]:
genre_enc.loc[genre_enc['TVThrillers'] == 1, 'Thrillers'] = 1
genre_enc = genre_enc.drop('TVThrillers', axis=1)

In [152]:
genre_enc.columns

Index(['Action', 'Anime', 'British', 'Family', 'Classic', 'Comedy', 'Crime',
       'Cult', 'Documentaries', 'Dramas', 'Faith', 'Horror', 'Independant',
       'International', 'Kids', 'Korean', 'LGBTQ', 'Musical', 'Reality',
       'Romance', 'SciFiFan', 'Science', 'Spanish', 'Sports', 'TalkShow',
       'TVComedies', 'Mystery', 'Teen', 'Thrillers'],
      dtype='object')

In [153]:
df_enc = pd.concat([df, rating_enc, type_enc, genre_enc], axis=1)
df_enc = df_enc.drop(['rating', 'type', 'category'], axis=1)

In [154]:
df_enc

id                  title  year  \
0        s1   Dick Johnson Is Dead  2020   
1        s2          Blood & Water  2021   
2        s3              Ganglands  2021   
3        s4  Jailbirds New Orleans  2021   
4        s5           Kota Factory  2021   
...     ...                    ...   ...   
8802  s8803                 Zodiac  2007   
8803  s8804            Zombie Dumb  2018   
8804  s8805             Zombieland  2009   
8805  s8806                   Zoom  2006   
8806  s8807                 Zubaan  2015   

                                            description  Rated G  Rated NC-17  \
0     As her father nears the end of his life, filmm...        0            0   
1     After crossing paths at a party, a Cape Town t...        0            0   
2     To protect his family from a powerful drug lor...        0            0   
3     Feuds, flirtations and toilet talk go down amo...        0            0   
4     In a city of coaching centers known to train I...        0            0   
...                                                 ...      ...          ...   
8802  A political cartoonist, a crime reporter and a...        0            0   
8803  While living alone in a spooky town, a young g...        0            0   
8804  Looking to survive in a world taken over by zo...        0            0   
8805  Dragged from civilian life, a former superhero...        0            0   
8806  A scrappy but poor boy worms his way into a ty...        0            0   

      Rated PG  rating_PG-13  Rated R  Rated TV_14  Rated TV_G  Rated TV_MA  \
0            0             1        0            0           0            0   
1            0             0        0            0           0            1   
2            0             0        0            0           0            1   
3            0             0        0            0           0            1   
4            0             0        0            0           0            1   
...        ...           ...      ...          ...         ...          ...   
8802         0             0        1            0           0            0   
8803         0             0        0            0           0            0   
8804         0             0        1            0           0            0   
8805         1             0        0            0           0            0   
8806         0             0        0            1           0            0   

      Rated TV_PG  Rated TV_Y  Rated TY_Y7  Rated TV_Y7_FV  Movie  Show  \
0               0           0            0               0      1     0   
1               0           0            0               0      0     1   
2               0           0            0               0      0     1   
3               0           0            0               0      0     1   
4               0           0            0               0      0     1   
...           ...         ...          ...             ...    ...   ...   
8802            0           0            0               0      1     0   
8803            0           0            1               0      0     1   
8804            0           0            0               0      1     0   
8805            0           0            0               0      1     0   
8806            0           0            0               0      1     0   

      Action  Anime  British  Family  Classic  Comedy  Crime  Cult  \
0          0      0        0       0        0       0      0     0   
1          0      0        0       0        0       0      0     0   
2          1      0        0       0        0       0      1     0   
3          0      0        0       0        0       0      0     0   
4          0      0        0       0        0       0      0     0   
...      ...    ...      ...     ...      ...     ...    ...   ...   
8802       0      0        0       0        0       0      0     1   
8803       0      0        0       0        0       0      0     0   
8804       0      0        

# Recommendation System

In [155]:
cosine_sim = cosine_similarity(df_enc.drop(['id','title','year','description'],axis=1))

In [156]:
def get_recommendations(title_list, cosine_sim_matrix, df, top_n=10):
    avg_sim_scores = np.zeros(len(df))

    for title in title_list:
        # Check if the title exists in the dataset
        if title not in df['title'].values:
            print("Warning: '{}' not found in the dataset.".format(title))
            continue
        
        # Get the index of the title in the DataFrame
        idx = df[df['title'] == title].index[0]
        sim_scores = cosine_sim_matrix[idx]
        
        # Add the similarity scores to the average scores array
        avg_sim_scores += sim_scores

    # Average the similarity scores across all titles in the input list
    avg_sim_scores /= len(title_list)
    movie_indices = np.argsort(avg_sim_scores)[::-1][:top_n]

    recommendations = df.iloc[movie_indices]['title']

    return recommendations

movie_titles = ["Henry Danger", "Inception", "Stranger Things"]
recommendations = get_recommendations(movie_titles, cosine_sim, df)
print(recommendations)

1465                       The Haunted Hathaways
1402                              Disenchantment
1473              Chilling Adventures of Sabrina
3685                             Stranger Things
2190                        The Umbrella Academy
7847                                Red vs. Blue
7551                Mystery Science Theater 3000
4368    Mystery Science Theater 3000: The Return
5650                  Star Trek: Deep Space Nine
1779             Marvel's Agents of S.H.I.E.L.D.
Name: title, dtype: object


In [207]:
get_recommendations(["Breaking Bad"],cosine_sim,df_enc)

5940                           Breaking Bad
3762                    Designated Survivor
678     The Assassination of Gianni Versace
1477                                Dare Me
2767                                  Ozark
6137                         American Crime
3065                                Messiah
5029                          Seven Seconds
3801                              WHAT / IF
1933                              Wentworth
Name: title, dtype: object

In [205]:
df_enc[df_enc['title'].str.contains("Spider-Man")]

id                                   title  year  \
1253  s1254  LEGO Marvel Spider-Man: Vexed by Venom  2019   
8067  s8068                            Spider-Man 3  2007   
8068  s8069       Spider-Man: Into the Spider-Verse  2018   

                                            description  Rated G  Rated NC-17  \
1253  When Green Goblin and Venom team up for a disa...        0            0   
8067  The seemingly invincible Spider-Man goes up ag...        0            0   
8068  After being bitten by a radioactive spider, Br...        0            0   

      Rated PG  rating_PG-13  Rated R  Rated TV_14  Rated TV_G  Rated TV_MA  \
1253         0             0        0            0           0            0   
8067         0             1        0            0           0            0   
8068         1             0        0            0           0            0   

      Rated TV_PG  Rated TV_Y  Rated TY_Y7  Rated TV_Y7_FV  Movie  Show  \
1253            0           1            0               0      1     0   
8067            0           0            0               0      1     0   
8068            0           0            0               0      1     0   

      Action  Anime  British  Family  Classic  Comedy  Crime  Cult  \
1253       0      0        0       1        0       1      0     0   
8067       1      0        0       0        0       0      0     0   
8068       1      0        0       0        0       1      0     0   

      Documentaries  Dramas  Faith  Horror  Independant  International  Kids  \
1253              0       0      0       0            0              0     1   
8067              0       0      0       0            0              0     0   
8068              0       0      0       0            0              0     0   

      Korean  LGBTQ  Musical  Reality  Romance  SciFiFan  Science  Spanish  \
1253       0      0        0        0        0         0        0        0   
8067       0      0        0        0        0         1        0        0   
8068       0      0        0        0        0         0        0        0   

      Sports  TalkShow  TVComedies  Mystery  Teen  Thrillers  
1253       0         0           0        0     0          0  
8067       0         0           0        0     0          0  
8068       0         0           0        0     0          0